assets/blocks/reader-registration/view.js

```diff
@@ -53,5 +65,45 @@ import './style.scss';
 				res.json().then( ( { message, data } ) => endLoginFlow( message, res.status, data ) );
 			} );
 		} );
+
+		const googleLogin = container.querySelector( '#newspack-google-login' );
+		if ( googleLogin ) {
+			googleLogin.onclick = () => {
+				startLoginFlow();
+				const checkLoginStatus = () => {
+					fetch( '/wp-json/newspack/v1/login/google/register', {
+						headers: new Headers( {
+							'X-WP-Nonce': window.newspack_reader_activation_data.nonce,
+						} ),
+					} ).then( res => {
+						res.json().then( ( { message, data } ) => endLoginFlow( message, res.status, data ) );
+					} );
+				};
+				fetch( '/wp-json/newspack/v1/login/google' )
+					.then( res => res.json().then( data => Promise.resolve( { data, status: res.status } ) ) )
+					.then( ( { data, status } ) => {
+						if ( status !== 200 ) {
+							endLoginFlow( data.message, status );
+						} else {
+							const authWindow = window.open(
+								'about:blank',
+								'newspack_google_login',
+								'width=500,height=600'
+							);
+							if ( authWindow ) {
+								authWindow.location = data;
+								const interval = setInterval( () => {
+									if ( authWindow.closed ) {
+										checkLoginStatus();
+										clearInterval( interval );
+									}
+								}, 500 );
+							} else {
+								endLoginFlow();
+							}
+						}
+					} );
```

Comment:
This means the SSO buttons should move inside the registration form, right?

Reply:
1. The form and 3rd party login are fundamentally different flows. The form submission handler expects an email in the payload, which can't be supplied by the 3rd party flow, because it's not user input. 
2. Any side-effect, such as newsletter subscription, should be done with WP hooks. I've updated the code here to handle newsletter lists signup in bcb43117c0e9cfac71eea7f9b6b4db88645fde43, with a [corresponding change in Newsletters](https://github.com/Automattic/newspack-newsletters/pull/895)

Reply:
After a more detailed discussion, I now agree that this is a good approach! 👍